### PR TITLE
Partial Solution: Accept host constraints in the cook json api

### PR DESCRIPTION
**Pull request by shamsimam**

### CODE REVIEW

File: scheduler/docs/scheduler-rest-api.asc [hunk 1]
```diff
@@ -75,6 +75,7 @@ The request body is list of job maps with each entry containing a map of the fol
 |`mem` | double | MB of requested memory.
 |`uris` | list of URI objects | A list of URIs that will be fetched into the container before launch.
 |`env` | JSON map | A map of environment variables to be provided to the job.
+|`host_constraints` | JSON map | A map of host constraints to be provided to the job.
```
Comment (dgrnbrg) [inline scheduler/docs/scheduler-rest-api.asc]: Once we cement the semantics, we'll need to make this more descriptive, and possibly include it in its own section. Also, we should indicate somewhere that it's optional.

File: scheduler/test/cook/test/mesos/api.clj [hunk 2]
```diff
@@ -108,14 +108,17 @@
...
   (let [conn (restore-fresh-database! "datomic:mem://mesos-api-test")
-        job (fn [cpus mem] {"uuid" (str (java.util.UUID/randomUUID))
-                            "command" "hello world"
-                            "name" "my-cool-job"
-                            "priority" 66
-                            "max_retries" 100
-                            "max_runtime" 1000000
-                            "cpus" cpus
-                            "mem" mem})
+        job-with-constraints (fn [cpus mem host-constraints]
```
Comment (dgrnbrg) [inline scheduler/test/cook/test/mesos/api.clj]: you can do:

```clojure
(fn inner-name
  ([x y] (inner-name x y 0))
  ([x y z] (+ x y z)))
```

**Comment by dgrnbrg** [comment]
Soon this will be something we can extend once Fenzo is merged to work end-to-end.

**Comment by dposada** [comment]
@shamsimam Is this still active or can we close?

**Comment by shamsimam** [comment]
I think this PR is now outdated and not required since we have Fenzo integration already working.

Closing.

**Pull request closed without merging**